libs/utils/analysis/latency_analysis.py

```diff
@@ -775,6 +776,26 @@ def plotRuntimes(self, task, tag=None, threshold_ms=8, bins=64):
         return stats_df.append(pd.DataFrame(
             stats.values(), columns=['running_time'], index=stats.keys()))
 
+    def plotTaskResidency(self, task):
+        """
+        Plot CPU residency of the specified task
+
+        :param task: Task PID
+        :type task: int
+        """
+        df = self._dfg_task_residency(task)
+
+        ax = df.plot(kind='bar', figsize=(16, 6))
+        ax.set_title('CPU residency of task {}'.format(task))
+
+        figname = os.path.join(
+            self._trace.plots_dir,
+            'task_cpu_residency_{}.png'.format(
+                self._trace.plots_prefix, task
+            )
+        )
```

Comment:
You could use the same approach as for the other functions, i.e.:

```python
figname = '{}/{}task_cpu_residency_{}.png'.format(self._trace.plots_dir, self._trace.plots_prefix, task)
```

Actually, you were missing the `plots_prefix` in the name of the file :)

Reply:
Woopsy daisy, you're right. Although I think for the path generation, `os.path.join`is preferred.